internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -317,9 +329,27 @@ func (h *PolicyChangeHandler) handlePolicyChange(ctx context.Context, c *config.
 		return fmt.Errorf("applying FleetClientConfig: %w", err)
 	}
 
+	if loggingHasChanged {
+		// The only way to update the logging configuration
+		// is to re-start the Elastic-Agent
+		h.coordinator.ReExec(nil)
```

Comment:
If you exposed EVENT_LOGGING_TO_STDERR and EVENT_LOGGING_TO_FILES as environment variables would this code even be necessary? On k8s this would automatically restart the container on every change.

The major con would be that changing these wouldn't work from Fleet outside of the container use case I suppose.

Reply:
Yes, this code is for the Fleet case. This PR addresses both cases: Fleet-managed agents and agents running in container, both standalone and fleet managed.

`EVENTS_TO_STDERR` is already exposed in the container command, the advantage of this single one over `EVENT_LOGGING_TO_STDERR` and `EVENT_LOGGING_TO_FILES` it prevents user error of configuring those two in a way that's invalid.


But to clarify, are you referring to the container case or a non-container execution of the agent?

Reply:
When I made this comment, I was thinking only of the container use case. Then I realized it also handles Fleet which we want to keep. Marking this as resolved.